### PR TITLE
refactored queries, ante/post handlers and DenomResolver

### PR DESCRIPTION
**Pull request by swelf19**
- Fields/structures/queries and all entities that are essentially gas_price but were named fee, renamed to gas_price
- Extended `DenomResolver` interface, the interface requires an additional method `ExtraDenoms`, the method should return a list of all denoms in which it is possible to pay fees. Used in GasPrices method to display gas prices for all currencies.
- The `ConvertToDenom` `DenomResolver` method now works with DecCoin instead of Coin.
- Params.Enabled parameter turns off not only post handler, but also ante handler.
- Query server refactored. Now it serves two queries.
    1) `GasPrice` with a denom payload as argument. Returns a gas price for specified denom
    2)  `GasPrices` without arguments. Returns a list of all gas prices (addetionel denoms come from `ExtraDenoms` method). Those can be used as fees

### CODE REVIEW

File: x/feemarket/ante/fee.go [hunk 1]
```diff
@@ -60,48 +53,51 @@ func (dfd FeeMarketCheckDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 		return ctx, errorsmod.Wrapf(feemarkettypes.ErrTooManyFeeCoins, "got length %d", len(feeCoins))
 	}
 
+	feeCoin := feeCoins[0]
+	feeGas := int64(feeTx.GetGas())
+
+	minGasPrice, err := dfd.feemarketKeeper.GetMinGasPrice(ctx, feeCoins[0].GetDenom())
```
Comment (aljo242) [inline x/feemarket/ante/fee.go:59]:
```suggestion
	minGasPrice, err := dfd.feemarketKeeper.GetMinGasPrice(ctx, feeCoin.GetDenom())
```

File: x/feemarket/post/fee.go [hunk 2]
```diff
@@ -89,13 +86,21 @@ func (dfd FeeMarketDeductDecorator) PostHandle(ctx sdk.Context, tx sdk.Tx, simul
 		return ctx, errorsmod.Wrapf(feemarkettypes.ErrTooManyFeeCoins, "got length %d", len(feeCoins))
 	}
 
+	feeCoin := feeCoins[0]
+	feeGas := int64(feeTx.GetGas())
+
+	minGasPrice, err := dfd.feemarketKeeper.GetMinGasPrice(ctx, feeCoins[0].GetDenom())
```
Comment (aljo242) [inline x/feemarket/post/fee.go:92]:
```suggestion
	minGasPrice, err := dfd.feemarketKeeper.GetMinGasPrice(ctx, feeCoin.GetDenom())
```

File: x/feemarket/keeper/feemarket.go [hunk 3]
```diff
@@ -85,8 +108,26 @@ func (k *Keeper) GetMinGasPrices(ctx sdk.Context) (sdk.DecCoins, error) {
 		return sdk.NewDecCoins(), err
 	}
 
-	fee := sdk.NewDecCoinFromDec(params.FeeDenom, baseFee)
-	minGasPrices := sdk.NewDecCoins(fee)
+	minGasPrice := sdk.NewDecCoinFromDec(params.FeeDenom, baseGasPrice)
+	minGasPrices := sdk.NewDecCoins(minGasPrice)
+
+	extraDenoms, err := k.resolver.AllowedDenoms(ctx)
```
Comment (aljo242) [inline x/feemarket/keeper/feemarket.go:114]: Won't the base denom of the feemarket be included in `AllowedDenoms`?

File: tests/e2e/setup.go [hunk 4]
```diff
@@ -162,7 +162,7 @@ func (s *TestSuite) QueryBaseFee() sdk.DecCoins {
 	// create the oracle client
 	c := types.NewQueryClient(cc)
 
-	resp, err := c.BaseFee(context.Background(), &types.BaseFeeRequest{})
+	resp, err := c.BaseFee(context.Background(), &types.GasPriceRequest{})
```
Comment (aljo242) [inline tests/e2e/setup.go:165]: This query needs an argument no?

e2e function should be renamed from `BaseFee`

File: x/feemarket/ante/fee.go [hunk 5]
```diff
@@ -60,48 +63,41 @@ func (dfd FeeMarketCheckDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
...
 
+	feeCoin := feeCoins[0]
+	feeGas := int64(feeTx.GetGas())
+
+	minGasPrice, err := dfd.feemarketKeeper.GetMinGasPrice(ctx, feeCoin.GetDenom())
+	if err != nil {
+		return ctx, errorsmod.Wrapf(err, "unable to get min gas price for denom %s", feeCoin.GetDenom())
+	}
+
+	// TODO: ??? log all gas prices, not only gas price with a given denom?
```
Comment (aljo242) [inline x/feemarket/ante/fee.go:74]: I think this TODO is not necessary

File: x/feemarket/client/cli/query.go [hunk 6]
```diff
@@ -85,25 +85,27 @@ func GetStateCmd() *cobra.Command {
...
 
-// GetBaseFeeCmd returns the cli-command that queries the current feemarket base fee.
+// GetBaseFeeCmd returns the cli-command that queries the current feemarket gas price.
 func GetBaseFeeCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "base-fee",
-		Short: "Query for the current feemarket base fee",
-		Args:  cobra.NoArgs,
-		RunE: func(cmd *cobra.Command, _ []string) error {
+		Use:   "gas-price",
```
Comment (aljo242) [inline x/feemarket/client/cli/query.go:91]: We should also add a cli query for `gas-prices`

**Comment by swelf19** [comment]
I also updated output format from
```bash
$ neutrond q feemarket gas-price untrn
0.002500000000000000untrn
```
```bash
$ neutrond q feemarket gas-prices
0.002500000000000000untrn
```
to
```bash
$ ./build/neutrond q feemarket gas-price untrn
price:
  amount: "0.002500000000000000"
  denom: untrn
```
wdyt?
it also allows you to encode output as json

**Review by aljo242 — APPROVED** [review]

**Pull request merged**